topic-operator/src/main/java/io/strimzi/operator/topic/Config.java

```diff
@@ -179,6 +189,16 @@ private Value(String key, Type<? extends T> type, boolean required) {
     public static final Value<String> TLS_KEYSTORE_PASSWORD = new Value<>(TC_TLS_KEYSTORE_PASSWORD, STRING, "");
     /** The endpoint identification algorithm used by clients to validate server host name. The default value is https. Clients including client connections created by the broker for inter-broker communication verify that the broker host name matches the host name in the broker’s certificate. Disable server host name verification by setting to an empty string.**/
     public static final Value<String> TLS_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM = new Value<>(TC_TLS_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM, STRING, "HTTPS");
+    /** If SASL should be used to authenticate with */
+    public static final Value<String> SASL_ENABLED = new Value<>(TC_SASL_ENABLED, STRING, "false");
+    /** The SASL mechanism to be used */
+    public static final Value<String> SASL_MECHANISM = new Value<>(TC_SASL_MECHANISM, STRING, "");
+    /** The SASL username to be used for authentication */
+    public static final Value<String> SASL_USERNAME = new Value<>(TC_SASL_USERNAME, STRING, "");
+    /** The SASL password to be used for authentication */
+    public static final Value<String> SASL_PASSWORD = new Value<>(TC_SASL_PASSWORD, STRING, "");
+    /** The security protocol to be used */
+    public static final Value<String> SECURITY_PROTOCOL = new Value<>(TC_SECURITY_PROTOCOL, STRING, "PLAINTEXT");
```

Comment:
Adding the security protocol, which can take the values `PLAINTEXT`, `SSL`, `SASL_PLAINTEXT` and `SASL_SSL`, means there's some duplication between this and the existing `TLS_ENABLED` config. I think we should give `TLS_ENABLED` no default here and in `Session` we should:

1. If `SECURITY_PROTOCOL` and `TLS_ENABLED` are both given ensure they're consistent and use the `SECURITY_PROTOCOL`
2. If only `SECURITY_PROTOCOL` is given then use it.
3. If only `TLS_ENABLED` is enabled then use `SSL` or `PLAINTEXT`
4. If neither are given then use `SSL`

We could deprecate the `TLS_ENABLED` at the same time (but would continue to configure both `SECURITY_PROTOCOL` and `TLS_ENABLED` in the EntityTopicOperator for now). Wdyt @scholzj @sknot-rh?

Reply:
Should I proceed with this?

Reply:
Yes. @scholzj @sknot-rh can speak up if they disagree.